### PR TITLE
feat: Dev Preview Banner + card adjustments

### PR DESCRIPTION
**Pull request by gabewomble**
Fixes https://github.com/cdklabs/construct-hub-webapp/issues/238

This PR implements a dismiss-able banner to indicate the application is in Dev Preview. The dismissal lasts for the duration of a session via the SessionStorage API

Screen Shot 2021-07-26 at 11 45 12 AM

Additional Changes:
- Enforces consistent layout spacing on pages
- Card border & shadow updated to further match design
- Documentation borders & dividers match design colors
- Headings in markdown have addition top margin (except first)
- Adjusted Heading layout to prevent word-break on logo for tablets

### CODE REVIEW

File: src/components/DevPreviewBanner/DevPreviewBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+          color="white"
+          mt={4}
+          mx={4}
+          p={4}
+          position="relative"
+          role="alertdialog"
+        >
+          <InfoOutlineIcon h={5} left={4} position="absolute" top={4} w={5} />
+          <Text fontSize="md" mx={8} textAlign="center">
+            This application is in Developer Preview. Some features may change.
```
Comment (addihorowitz) [inline src/components/DevPreviewBanner/DevPreviewBanner.tsx:40]: Elad suggested we use "beta" which is more publicly used than "developer preview"

File: src/views/Package/components/PackageDocsError/PackageDocsError.tsx [hunk 2]
```diff
@@ -20,11 +20,12 @@ export const PackageDocsError: FunctionComponent<PackageDocsErrorProps> = ({
       align="center"
       fontSize="xl"
       fontStyle="oblique"
+      px={4}
       wordBreak="break-word"
     >
       Documentation in {language} is still not ready for this package. Come back
-      soon. If this issue persists, please let us know by creting an {issueLink}
-      .
+      soon. If this issue persists, please let us know by creating an{" "}
```
Comment (addihorowitz) [inline src/views/Package/components/PackageDocsError/PackageDocsError.tsx:27]: Is the additional {" "} intentional?

**Review by addihorowitz — COMMENTED** [review]
I left two comments

File: src/views/Package/components/PackageDocsError/PackageDocsError.tsx (hunk 2 above)
Comment (gabewomble) [reply src/views/Package/components/PackageDocsError/PackageDocsError.tsx:27]: Yep, this is to preserve the space between the text & the `<ExternalLink />` on the next line!

File: src/components/DevPreviewBanner/DevPreviewBanner.tsx (hunk 1 above)
Comment (gabewomble) [reply src/components/DevPreviewBanner/DevPreviewBanner.tsx:40]:
```suggestion
            This application is in Beta. Some features may change.
```

**Review by addihorowitz — APPROVED** [review]
Reviewed and approved.

**Pull request merged**